cfme/utils/targeted_refresh.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+class for targeted refresh tests
+"""
+
+import re
+import attr
+
+
+@attr.s
+class TargetedRefresh(object):
```

Comment:
What this class in fact it does is it checks specific log file and searches for a list of regular expressions to occur until stopped or until all regexes are found, with some timeout.

This is very useful feature IMO and we should make it reusable for arbitrary regular expressions, not lock it for targeted refresh only. Therefore I suggest we rename the class to

```
class LogFinder:
   ...
```

and allow the caller to use it for searching arbitrary regular expressions. We would then modify the caller, which is `targeted_refresh` fixture in our case, to form the correct regex, which is 

```python
r'^.*refresh_targets_for_ems.*Collection of targets with id.*:ems_ref=> "{}".*.$'.format('ems_ref')
```

in our case.

Wdyt?

Reply:
This is a great idea in IMO. I already thought about making it reusable for other cases but didn't know how. I think I will make a new PR for this.